chatterbot/adapters/plugins/evaluate_mathematically.py

```diff
@@ -0,0 +1,200 @@
+from .plugin import PluginAdapter
+import re
+import os, json
+
+class EvaluateMathematically(PluginAdapter):
+
+    def should_answer(self, input_text):
+        """
+        Determines whether it is appropriate for this plugin
+        to respond to the user input.
+        """
+
+        response = self.process( input_text )
+
+        if response == False:
+            return False
+        else:
+            return True
+
+
+    def process(self, input_text):
+        """
+        Takes a statement string.
+        Returns the simplified statement string
+        with the mathematical terms "solved".
+        """
+
+        # Getting the mathematical terms within the input statement
+        expression, string = self.simplify_chunks( self.normalize( input_text ) )
+
+        # Returning important information
+        try:
+            string += '= ' + str( eval( string ) )#self.evaluate( string ) )
+
+            return string
+        except:
+            return False
+
+
+    def simplify_chunks(self, input_text):
+        """
+        Separates the incoming text.
+        """
+
+        expression = []
+        string = ''
+
+        for chunk in input_text.split( ' ' ):
+
+            is_integer = self.isInteger( chunk )
+
+            if is_integer == False:
+                is_float = self.isFloat( chunk )
+
+                if is_float == False:
+                    is_operator = self.isOperator( chunk )
+
+                    if is_operator == False:
+                        continue
+                    else:
+                        expression.append( is_operator )
+
+                        string += str( is_operator ) + ' '
+                else:
+                    expression.append( is_float )
+
+                    string += str( is_float ) + ' '
+            else:
+                expression.append( is_integer )
+
+                string += str( is_integer ) + ' '
+
+        return expression, string
+
+
+    def evaluate( self, expression ):
+        """
+        Evaluates a set of expressions
+        and produces an answer. Then,
+        it returns the answer.
+        """
+
+        return eval( expression )
+
+
+    def isFloat(self, string):
+        """
+        If the string is a float, returns
+        the float of the string. Otherwise,
+        it returns False.
+        """
+
+        try:
+            float( integer )
```

Comment:
I honestly think the `try...except` block is a bit more clear here, and it avoids any edge cases with determining what is and isn't a float. If Python thinks it's a float, why argue with it?

You may want to try to coerce this to a `Decimal` though, that way you don't need to worry about floating point oddities.

```python
try:
    return decimal.Decimal(string)
except decimal.DecimalException:
    return False
```

If you want to be _really careful_, there are [other edge cases](https://github.com/tomchristie/django-rest-framework/blob/ff0eeaf77addc29d1ce0c10f47948ad06611f526/rest_framework/fields.py#L924-L936) that we handle in DRF for parsing decimal values.